src/types/client-manager.ts

```diff
@@ -130,6 +213,237 @@ export class ClientManager {
   public getTypedAppClientById<TClient>(typedClient: TypedAppClient<TClient>, details: TypedAppClientByIdDetails) {
     return new typedClient({ ...details, resolveBy: 'id' }, this._algod)
   }
+
+  /**
+   * Retrieve client configurations from environment variables when defined or get defaults (expects to be called from a Node.js environment)
+   *
+   * If `process.env.ALGOD_SERVER` is defined it will use that along with optional `process.env.ALGOD_PORT` and `process.env.ALGOD_TOKEN`.
+   *
+   * If `process.env.INDEXER_SERVER` is defined it will use that along with optional `process.env.INDEXER_PORT` and `process.env.INDEXER_TOKEN`.
+   *
+   * If either aren't defined it will use the default LocalNet config, noting if `process.env.ALGOD_SERVER` is specified, but
+   * `process.env.INDEXER_SERVER` isn't then it will leave indexer as `undefined`.
+   *
+   * It will return a KMD configuration that uses `process.env.KMD_PORT` (or port 4002) if `process.env.ALGOD_SERVER` is defined,
+   * otherwise it will use the default LocalNet config unless it detects testnet or mainnet.
+   * @example
+   * ```typescript
+   * const config = ClientManager.getConfigFromEnvironmentOrLocalNet()
+   * ```
+   * @returns The config for algod, indexer and kmd
+   */
+  public static getConfigFromEnvironmentOrLocalNet(): AlgoConfig {
+    if (!process || !process.env) {
+      throw new Error('Attempt to get default client configuration from a non Node.js context; supply the config instead')
+    }
+    const algodConfig = !process.env.ALGOD_SERVER
+      ? ClientManager.getDefaultLocalNetConfig('algod')
+      : ClientManager.getAlgodConfigFromEnvironment()
+
+    const indexerConfig = !process.env.INDEXER_SERVER
+      ? ClientManager.getDefaultLocalNetConfig('indexer')
+      : !process.env.ALGOD_SERVER
```

Comment:
This is technically a breaking change, but I figured it would be super weird if you used environment variables for algod, but were happy with default localnet for indexer / kmd so probably wouldn't break anyone in practice.

Given the concept of indexer and kmd being optional is more baked in now it felt better to make them undefined in the case you only define algod environment variables so you can support that explicitly.